### PR TITLE
Issue #958: Update manager not configured for layouts

### PR DESCRIPTION
**Pull request by docwilmot**
First pass at fixing https://github.com/backdrop/backdrop-issues/issues/958.
Issues with this patch: 
`system.updater.inc has a new`LayoutUpdater` class with the following which is a bit shaky:

```
  static function canUpdateDirectory($directory) {
    // This is a lousy test, but don't know how else to confirm it is a layout.
    if (file_scan_directory($directory, '/(?=.*template\.php$)|(?=.*\.module$)/')) {
      return FALSE;
    }
    elseif (file_scan_directory($directory, '/.*layout--.*/')) {
      return TRUE;
    }
    return FALSE;
  }
```

This would be easier if we got in https://github.com/backdrop/backdrop-issues/issues/960.

Also, in `update.compare.inc` theres a temporary function 

```
//temp todo
function _system_rebuild_layout_data() {
  // Find modules
  $layout_data = layout_get_layout_info();
  foreach($layout_data as $layout) {
    $name = $layout['name'];
    $layouts[$name] = new stdClass();
    $layouts[$name]->name = $name;
    $layouts[$name]->uri = $layout['path'] . '/' . $name . '.info';
    $layouts[$name]->info = $layout;
    $layouts[$name]->status = 1;
  }

  return $layouts;
}
```

for compatibility with `system_rebuild_theme_data()` and the module one. There may be a pre-existing function which I cant find. If not do we need to create this function? will just a stub like this (renamed to `_update__rebuild_layout_data()` maybe?) do for now?

### CODE REVIEW

**Comment by docwilmot** [comment]
OOps the changes to `admin_bar.map.inc` were not to have been in this PR. Apologies.

File: core/modules/system/system.updater.inc [hunk 1]
```diff
@@ -148,3 +153,64 @@ class ThemeUpdater extends Updater implements BackdropUpdaterInterface {
...
   }
 }
+
+/**
+ * Class for updating layouts using FileTransfer classes via authorize.php.
+ */
+class LayoutUpdater extends Updater implements BackdropUpdaterInterface {
+
+  /**
+   * Return the directory where a theme should be installed.
```
Comment (quicksketch) [inline core/modules/system/system.updater.inc:163]: theme => layout

File: core/modules/system/system.updater.inc [hunk 2]
```diff
@@ -3,7 +3,7 @@
 /**
  * @file
  * Subclasses of the Updater class to update Backdrop core knows how to update.
- * At this time, only modules and themes are supported.
+ * At this time, only modules, themes and layouts are supported.
```
Comment (quicksketch) [inline core/modules/system/system.updater.inc:6]: Could we add in an Oxford comma here and everywhere else? `modules, themes, and layouts`

File: core/modules/update/update.module [hunk 3]
```diff
@@ -158,7 +159,7 @@ function update_menu() {
   );
 
   // We want action links for updating projects at a few different locations:
-  // both the module and theme administration pages, and on the available
+  // on the Module, theme and layout administration pages, and on the available
```
Comment (quicksketch) [inline core/modules/update/update.module:162]: `Module` => `module` (and Oxford comma)

**Pull request closed without merging**